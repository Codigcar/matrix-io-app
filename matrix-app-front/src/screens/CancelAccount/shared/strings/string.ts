import { i18n } from 'src/utils/core/MTXStrings';

export const string = {
  // TODO: Move these to shared/strings when we have all changes of Dambert related to refactor
  appName: i18n.t('app-name'),
  // TODO END
  cancelAccountCompleteSecondSectionText1: i18n.t(
    'cancelAccount:cancelAccountComplete.second-section-text-1',
  ),
  cancelAccountCompleteSecondSectionText3: i18n.t(
    'cancelAccount:cancelAccountComplete.second-section-text-3',
  ),
  cancelAccountCompleteSecondSectionText2: i18n.t(
    'cancelAccount:cancelAccountComplete.second-section-text-2',
  ),
  cancelAccountCompleteFirstSectionTitle: i18n.t(
    'cancelAccount:cancelAccountComplete.first-section-title',
  ),
  cancelAccountCompleteSecondSectionTitle: i18n.t(
    'cancelAccount:cancelAccountComplete.second-section-title',
  ),
  cancelAccountCompleteThirdSectionText1: i18n.t(
    'cancelAccount:cancelAccountComplete.third-section-text-1',
  ),
  cancelAccountCompleteThirdSectionText2: i18n.t(
    'cancelAccount:cancelAccountComplete.third-section-text-2',
  ),
  cancelAccountCompleteThirdSectionText3: i18n.t(
    'cancelAccount:cancelAccountComplete.third-section-text-3',
  ),
  cancelAccountCompleteHeader: i18n.t('cancelAccount:cancelAccountComplete.header'),
  cancelAccountCompleteTitle: i18n.t('cancelAccount:cancelAccountComplete.title'),
  cancelAccountCompleteSubtitle: i18n.t('cancelAccount:cancelAccountComplete.subtitle'),
  cancelAccountCompleteChatButton: i18n.t('cancelAccount:cancelAccountComplete.chat-button'),
  cancelAccontCompletePaymentButton: i18n.t('cancelAccount:cancelAccountComplete.payment-button'),
  cancelAccountCompleteContinueButton: i18n.t(
    'cancelAccount:cancelAccountComplete.continue-button',
  ),
  cancelAccountSurveyModalTitle: i18n.t('cancelAccount:cancelAccount.survey.modal.title'),
  cancelAccountSurveyModalSubtitle1: i18n.t('cancelAccount:cancelAccount.survey.modal.subtitle-1'),
  cancelAccountSurveyModalSubtitle2: i18n.t('cancelAccount:cancelAccount.survey.modal.subtitle-2'),
  cancelAccountSurveyModalTextTitle: i18n.t('cancelAccount:cancelAccount.survey.modal.text-title'),
  cancelAccountSurveyModalTextSubtitle: i18n.t(
    'cancelAccount:cancelAccount.survey.modal.text-subtitle',
  ),
  cancelAccountSurveyModalCheckboxInfo: i18n.t(
    'cancelAccount:cancelAccount.survey.modal.checkbox-info',
  ),
  cancelAccountSurveyModalBtnCancelAccount: i18n.t(
    'cancelAccount:cancelAccount.survey.modal.btnCancelAccount',
  ),
  cancelAccountHeader: i18n.t('cancelAccount:cancelAccount.header'),
  cancelAccountSurveyTitle: i18n.t('cancelAccount:cancelAccount.survey.title'),
  cancelAccountSurveySubtitle: i18n.t('cancelAccount:cancelAccount.survey.subtitle'),
  cancelAccountSurveyQuestion: i18n.t('cancelAccount:cancelAccount.survey.question'),
  cancelAccountSurveyOption4Title: i18n.t('cancelAccount:cancelAccount.survey.option-4-title'),
  cancelAccountSurveyOption4Placeholder: i18n.t(
    'cancelAccount:cancelAccount.survey.option-4-placeholder',
  ),
  cancelAccountSurveyContinueButton: i18n.t('cancelAccount:cancelAccount.survey.continue-button'),
  cancelAccountSurveyOption1: i18n.t('cancelAccount:cancelAccount.survey.option-1'),
  cancelAccountSurveyOption2: i18n.t('cancelAccount:cancelAccount.survey.option-2'),
  cancelAccountSurveyOption3: i18n.t('cancelAccount:cancelAccount.survey.option-3'),
  cancelAccountSurveyOption4: i18n.t('cancelAccount:cancelAccount.survey.option-4'),
  cancelAccountTitle: i18n.t('cancelAccount:cancelAccount.title'),
  cancelAccountSubtitle1: i18n.t('cancelAccount:cancelAccount.subtitle1'),
  cancelAccountSubtitle2: i18n.t('cancelAccount:cancelAccount.subtitle2'),
  cancelAccountQuestionLabel: i18n.t('cancelAccount:cancelAccount.question-label'),
  cancelAccountContinueButton: i18n.t('cancelAccount:cancelAccount.continue-button'),
  cancelAccountWaitingErrorTitle: i18n.t(
    'cancelAccount:cancelAccount.cancel-account-waiting.errorTitle',
  ),
  cancelAccountWaitingErrorSubtitle: i18n.t(
    'cancelAccount:cancelAccount.cancel-account-waiting.errorSubtitle',
  ),
  cancelAccountWaitingErrorDescription: i18n.t(
    'cancelAccount:cancelAccount.cancel-account-waiting.errorDescription',
  ),
  cancelAccountWaitingErrorButtonLabel: i18n.t(
    'cancelAccount:cancelAccount.cancel-account-waiting.errorButtonLabel',
  ),
  cancelAccountWaitingWaitingTitle: i18n.t(
    'cancelAccount:cancelAccount.cancel-account-waiting.waiting-title',
  ),
  cancelAccountWaitingWaitingSubtitle1: i18n.t(
    'cancelAccount:cancelAccount.cancel-account-waiting.waiting-subtitle-1',
  ),
  cancelAccountWaitingWaitingSubtitle2: i18n.t(
    'cancelAccount:cancelAccount.cancel-account-waiting.waiting-subtitle-2',
  ),
};
