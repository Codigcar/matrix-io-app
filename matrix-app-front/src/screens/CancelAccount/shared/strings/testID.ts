export const testID = {
  iconListItem: 'iconListItemId',
  dataListComplete: 'dataListCompleteId',
  cancelAccountSurveyActionButton: 'cancelAccountSurveyActionButtonId',
  cancelAccountSurveyModal: 'cancelAccountSurveyModalId',
};
