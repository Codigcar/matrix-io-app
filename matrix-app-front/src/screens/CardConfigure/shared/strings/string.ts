import { i18n } from 'src/utils/core/MTXStrings';

export const string = {
  activated: i18n.t('cardConfigure:status.activated'),
  desactivated: i18n.t('cardConfigure:status.deactivated'),
  noCard: i18n.t('cardConfigure:status.no-card'),
  cardBlocked: i18n.t('cardConfigure:status.card-blocked'),
  activatedPurchases: i18n.t('cardConfigure:status.activated-purchases'),
  desactivatedPurchases: i18n.t('cardConfigure:status.deactivated-purchases'),
  configureCardSubtitle: i18n.t('cardConfigure:home.configure-card-0-subtitle'),
  configureCardTitle: i18n.t('cardConfigure:home.configure-card-title'),
  configureCard0Title: i18n.t('cardConfigure:home.configure-card-0-title'),
  configureCard0SubtitleInit: i18n.t('cardConfigure:home.configure-card-0-subtitle-init'),
  walletFlowNFCIncopatibleTitle: i18n.t('cardConfigure:wallet-flow.nfc.incompatible.title'),
  walletFlowNFCIncopatibleMessage: i18n.t('cardConfigure:wallet-flow.nfc.incompatible.message'),
  walletFlowUnderstood: i18n.t('cardConfigure:wallet-flow.understood'),
  cardConfigurationVirtualCard: i18n.t('cardConfigure:virtual-card'),
  cardConfigurationPhysicalCard: i18n.t('cardConfigure:physical-card'),
  cardConfigurationSupplementaryCardLabel: i18n.t('cardConfigure:supplementary-card.label'),
  cardConfigurationSupplementaryCardHelper: i18n.t('cardConfigure:supplementary-card.helper'),
  cardConfigurationSupplementaryCardMessage: i18n.t('cardConfigure:supplementary-card.message'),
  cardConfigurationSupplementaryCardActionText: i18n.t(
    'cardConfigure:supplementary-card.action-text',
  ),
  configureCard0Detail0: i18n.t('cardConfigure:home.configure-card-0-detail-0'),
  configureCard1Detail0: i18n.t('cardConfigure:home.configure-card-1-detail-0'),
  configureCard0Detail1: i18n.t('cardConfigure:home.configure-card-0-detail-1'),
  configureCard1Detail1: i18n.t('cardConfigure:home.configure-card-1-detail-1'),
  configureCardError01: i18n.t('cardConfigure:home.configure-card-error-01'),
  configureCardError02: i18n.t('cardConfigure:home.configure-card-error-02'),
  configureCardInternetDesactivated: i18n.t(
    'cardConfigure:home.configure-card-internet-deactivated',
  ),
  configureCardInternetActivated: i18n.t('cardConfigure:home.configure-card-internet-activated'),
  configureCardForeignDesactivated: i18n.t('cardConfigure:home.configure-card-foreign-deactivated'),
  configureCardForeignActivated: i18n.t('cardConfigure:home.configure-card-foreign-activated'),
};
