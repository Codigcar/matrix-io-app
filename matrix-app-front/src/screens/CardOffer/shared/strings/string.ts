import { i18n } from 'src/utils/core/MTXStrings';

export const string = {
  cardOfferDocumentsTitle: i18n.t('cardOffer:card-offer-documents-title'),
  cardOfferDocumentsSubtitle: i18n.t('cardOffer:card-offer-documents-subtitle'),
  cardOfferDocumentContractTitle: i18n.t('cardOffer:card-offer-document-contract-title'),
  cardOfferDocumentResumeTitle: i18n.t('cardOffer:card-offer-document-resume-title'),
  cardOfferDocumentInsuranceTitle: i18n.t('cardOffer:card-offer-document-insurance-title'),
  cardOfferCompleteTitle: i18n.t('cardOffer:card-offer-complete-title'),
  cardOfferCompleteSubtitle: i18n.t('cardOffer:card-offer-complete-subtitle'),
  buttonLabelContinue: i18n.t('button-label-continue'),

  cardOfferModalsWarningAuthorizationTitle: i18n.t('cardOffer:card-offer-modals.warning-authorization-title'),
  cardOfferModalsErrorAuthorizationTitle: i18n.t('cardOffer:card-offer-modals.error-authorization-title'),
  cardOfferModalsWarningAuthorizationSubtitle: i18n.t('cardOffer:card-offer-modals.warning-authorization-subtitle'),
  cardOfferModalsErrorAuthorizationSubtitle: i18n.t('cardOffer:card-offer-modals.error-authorization-subtitle'),
  cardOfferModalsWarningAuthorizationButton: i18n.t('cardOffer:card-offer-modals.warning-authorization-button'),
  cardOfferModalsErrorAuthorizationButton: i18n.t('cardOffer:card-offer-modals.error-authorization-button'),

  cardOfferModalsLoadingOfferTitle: i18n.t('cardOffer:card-offer-modals.loading-offer-title'),
  cardReissueLoadGeneratingCardTitle: i18n.t('cardReissue.load.generating-card-title'),
  cardOfferModalsLoadingAuthorizationTitle: i18n.t('cardOffer:card-offer-modals.loading-authorization-title'),
  cardOfferModalsLoadingOfferSubtitle: i18n.t('cardOffer:card-offer-modals.loading-offer-subtitle'),
  cardReissueLoadGeneratingCardSubtitle: i18n.t('cardReissue.load.generating-card-subtitle'),
  cardOfferModalsLoadingAuthorizationSubtitle: i18n.t('cardOffer:card-offer-modals.loading-authorization-subtitle'),

  noCardOfferMessage: i18n.t('cardOffer:no-card-offer-message'),
  noCardOfferTitle: i18n.t('cardOffer:no-card-offer-title'),
  noCardOfferSubtitle: i18n.t('cardOffer:no-card-offer-subtitle'),
  finish: i18n.t('finish'),

  cardOfferBenefitListFirstItemTitle: i18n.t('card-offer-benefit-list-first-item-title'),
  cardOfferBenefitListSecondItemTitle: i18n.t('card-offer-benefit-list-second-item-title'),
  cardOfferBenefitListThirdItemTitle: i18n.t('card-offer-benefit-list-third-item-title'),
  cardOfferBenefitListFourthItemTitle: i18n.t('card-offer-benefit-list-fourth-item-title'),

  cardOfferBenefitFirstListSectionTitle: i18n.t('cardOffer:card-offer-benefit-first-list-section-title'),
  cardOfferBenefitSecondListSectionLink: i18n.t('cardOffer:card-offer-benefit-second-list-section-link'),
  cardOfferBenefitSecondListSectionTitle: i18n.t('cardOffer:card-offer-benefit-second-list-section-title'),
  understood: i18n.t('understood'),

  cardOfferBenefitFirstListCompleteSectionTitle: i18n.t('cardOffer:card-offer-benefit-first-list-complete-section-title'),
  cardOfferBenefitSecondListCompleteSectionTitle: i18n.t('cardOffer:card-offer-benefit-second-list-complete-section-title'),

  cardOfferTitle: i18n.t('cardOffer:card-offer-title'),
  cardOfferTitleLast: i18n.t('cardOffer:card-offer-title-last'),
  appName: i18n.t('app-name'),
  cardOfferAmountText: i18n.t('cardOffer:card-offer-amount-text'),
  cardOfferContractFirstText: i18n.t('cardOffer:card-offer-contract-first-text'),
  cardOfferContractLinkText: i18n.t('cardOffer:card-offer-contract-link-text'),
  cardOfferButtonLabel: i18n.t('cardOffer:card-offer-button-label'),
};
