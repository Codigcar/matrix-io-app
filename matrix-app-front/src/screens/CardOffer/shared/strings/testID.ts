export const testID = {
  backArrowBoxId: 'backArrowBox',
  shareIconBoxId: 'shareIconBox',
  shareIconId: 'shareIcon',

  documentListItemId: 'documentListItem',
  viewDocumentButtonMultipleId: 'viewDocumentButton',
  viewDocumentButtonId: 'viewDocumentButton',

  continueButtonId: 'continueButton',
  buttonWarningId: 'button-warning',
  btnShowInfoId: 'btn-show-info',
  btnUnderstoodId: 'btn-understood',
  closeButtonId: 'close-button',
  warningModalId: 'warning-modal',
  errorModalId: 'error-modal',
  btnTermsId: 'btn-terms',
  btnContinueGetOffer: 'btn-continue-get-offer',
  loadingModal: 'loading-modal',
};
