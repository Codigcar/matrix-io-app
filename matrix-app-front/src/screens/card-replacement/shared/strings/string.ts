import { i18n } from 'src/utils/core/MTXStrings';

export const string = {
  cardReplacementLoadingModalSubTitle: i18n.t('cardReplacement:loading-modal.subtitle'),
  cardReplacementLoadingModalMessage: i18n.t('cardReplacement:loading-modal.message'),
  cardReplacementValidationSuccessTitle: i18n.t('cardReplacement:validation-success.title'),
  cardReplacementValidationSuccessContinueMessage: i18n.t(
    'cardReplacement:validation-success.continue-message',
  ),
  cardReplacementValidationSuccessButton: i18n.t('cardReplacement:validation-success.button'),
  cardReplacementValidationErrorTitle: i18n.t('cardReplacement:validation-error.title'),
  cardReplacementValidationErrorSubtitle: i18n.t('cardReplacement:validation-error.subtitle'),
  cardReplacementValidationErrorMessage: i18n.t('cardReplacement:validation-error.message'),
  cardReplacementValidationErrorButton: i18n.t('cardReplacement:validation-error.button'),
  cardOfferBenefitListFirstItemTitle: i18n.t('card-offer-benefit-list-first-item-title'), // Stay in src/share
  cardOfferBenefitListSecondItemTitle: i18n.t('card-offer-benefit-list-second-item-title'), // Stay in src/share
  cardOfferBenefitListThirdItemTitle: i18n.t('card-offer-benefit-list-third-item-title'), // Stay in src/share
  cardOfferBenefitListFourthItemTitle: i18n.t('card-offer-benefit-list-fourth-item-title'), // Stay in src/share
  cardReplacementSummaryOfferBenefits: i18n.t('cardReplacement:summary-offer.benefits'),
  cardReplacementSummaryOfferCongrats: i18n.t('cardReplacement:summary-offer.congrats'),
  cardReplacementSummaryOfferSubtitle: i18n.t('cardReplacement:summary-offer.subtitle'),
  cardReplacementSummaryOfferFeedbackMessage: i18n.t(
    'cardReplacement:summary-offer.feedback-message',
  ),
  cardReplacementSummaryOfferGoHome: i18n.t('cardReplacement:summary-offer.go-home'),
};
