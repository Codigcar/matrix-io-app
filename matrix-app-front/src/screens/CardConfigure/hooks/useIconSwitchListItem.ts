import { string } from '../shared/strings/string';
import { IconSwitchListItemProps } from '../shared/types/components';

const useIconSwitchListItem = (props: IconSwitchListItemProps) => {
  const { isBlocked, status, type } = props;

  // Methods
  const changeText = () => {
    if (!isBlocked) {
      if (status) {
        return type === 'card' ? string.activated : string.activatedPurchases;
      }
      return type === 'card' ? string.desactivated : string.desactivatedPurchases;
    }
    return string.cardBlocked;
  };

  const changeColorStatus = () => {
    if (!isBlocked) {
      if (status) return 'complementaryMint700';
      return 'primary500';
    }
    return 'FeedbackError600';
  };

  return {
    changeText,
    changeColorStatus,
  };
};

export default useIconSwitchListItem;
