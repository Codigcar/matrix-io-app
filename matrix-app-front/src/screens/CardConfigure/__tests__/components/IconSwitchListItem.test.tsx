import React from 'react';
import { fireEvent, render, renderHook } from 'src/matrix-ui-components/utils/test-utils';
import { ForeignPurchaseRectable, InternetPurchaseRectangle } from 'assets/svgs';
import { testID } from '../../shared/strings/testID';
import IconSwitchListItem, { iconSelect } from '../../components/IconSwitchListItem';
import useIconSwitchListItem from '../../hooks/useIconSwitchListItem';
import { string } from '../../shared/strings/string';

interface ComponentRenderProps {
  loading?: boolean;
  isBlocked?: boolean;
  label?: string;
  onChange?: () => void;
}

interface ComponentIconSelectProps {
  type: string;
}

const mainComponentTestId = testID.iconSwitchListItem;

const ComponentRender = (props?: ComponentRenderProps) =>
  render(
    <IconSwitchListItem
      loading={props?.loading ?? false}
      isBlocked={props?.isBlocked ?? false}
      testID={mainComponentTestId}
      label={props?.label ?? 'label'}
      onChange={props?.onChange ?? jest.fn()}
    />,
  );

const ComponentIconSelect = (props: ComponentIconSelectProps) =>
  render(<>{iconSelect(props.type)}</>);
const ComponentForeignPurchaseRectable = () => render(<ForeignPurchaseRectable />);
const ComponentInternetPurchaseRectangle = () => render(<InternetPurchaseRectangle />);

describe('IconSwitchListItem component', () => {
  it('should render without errors', () => {
    const { getByTestId } = ComponentRender();
    const component = getByTestId(mainComponentTestId);
    expect(component).toBeTruthy();
  });

  it('should render loading switch', () => {
    const { getByTestId } = ComponentRender({ loading: true });
    const component = getByTestId(testID.iconSwitchListItemLoadingswitch);
    expect(component).toBeTruthy();
  });

  it('should render switch', () => {
    const { getByTestId } = ComponentRender();
    const component = getByTestId(testID.iconSwitchListItemSwitch);
    expect(component).toBeTruthy();
  });

  it('should render label', () => {
    const { getByText } = ComponentRender();
    const component = getByText('label');
    expect(component).toBeTruthy();
  });

  it('should be blocked', () => {
    const { getByTestId } = ComponentRender({ isBlocked: true });
    const component = getByTestId(testID.iconSwitchListItemSwitch);
    expect(component).toHaveAccessibilityState({ disabled: true });
  });

  it('should press switch', () => {
    const onChange = jest.fn();
    const { getByTestId } = ComponentRender({ onChange });
    const component = getByTestId(testID.iconSwitchListItemSwitch);
    fireEvent.press(component);
    expect(onChange).toHaveBeenCalled();
  });

  it('should render ComponentIconSelect correctly', () => {
    const internetComponent = ComponentIconSelect({ type: 'internet' });
    const foreignComponent = ComponentIconSelect({ type: 'foreign' });
    expect(internetComponent).toBeTruthy();
    expect(foreignComponent).toBeTruthy();
  });

  it('should render ComponentForeignPurchaseRectable correctly', () => {
    const component = ComponentForeignPurchaseRectable();
    expect(component).toBeTruthy();
  });

  it('should render ComponentInternetPurchaseRectangle correctly', () => {
    const component = ComponentInternetPurchaseRectangle();
    expect(component).toBeTruthy();
  });

  it('should render hook correctly', () => {
    const { result } = renderHook(() =>
      useIconSwitchListItem({
        label: 'label',
        onChange: jest.fn(),
        loading: false,
        isBlocked: false,
        status: false,
        type: 'card',
      }));

    expect(result.current.changeText()).toBe(string.desactivated);
    expect(result.current.changeColorStatus()).toBe('primary500');
  });

  it('should render hook correctly wit other params', () => {
    const { result } = renderHook(() =>
      useIconSwitchListItem({
        label: 'label',
        onChange: jest.fn(),
        loading: false,
        isBlocked: false,
        status: true,
        type: 'card',
      }));

    expect(result.current.changeText()).toBe(string.activated);
    expect(result.current.changeColorStatus()).toBe('complementaryMint700');
  });

  it('should render hook correctly wit other params (status false)', () => {
    const { result } = renderHook(() =>
      useIconSwitchListItem({
        label: 'label',
        onChange: jest.fn(),
        loading: false,
        isBlocked: false,
        status: false,
        type: 'other',
      }));

    expect(result.current.changeText()).toBe(string.desactivatedPurchases);
    expect(result.current.changeColorStatus()).toBe('primary500');
  });

  it('should render hook correctly wit other params (status true and other type)', () => {
    const { result } = renderHook(() =>
      useIconSwitchListItem({
        label: 'label',
        onChange: jest.fn(),
        loading: false,
        isBlocked: false,
        status: true,
        type: 'other',
      }));

    expect(result.current.changeText()).toBe(string.activatedPurchases);
    expect(result.current.changeColorStatus()).toBe('complementaryMint700');
  });

  it('should render hook correctly wit other params (status false with type card)', () => {
    const { result } = renderHook(() =>
      useIconSwitchListItem({
        label: 'label',
        onChange: jest.fn(),
        loading: false,
        isBlocked: false,
        status: false,
        type: 'card',
      }));

    expect(result.current.changeText()).toBe(string.desactivated);
    expect(result.current.changeColorStatus()).toBe('primary500');
  });

  it('should match snapshot', () => {
    const { toJSON } = ComponentRender();
    expect(toJSON()).toMatchSnapshot();
  });
});
