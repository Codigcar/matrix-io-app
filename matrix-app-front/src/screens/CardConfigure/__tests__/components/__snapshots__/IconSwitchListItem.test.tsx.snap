// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconSwitchListItem component should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
    ]
  }
  testID="iconSwitchListItemId"
>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "justifyContent": "flex-start",
            "marginTop": 4,
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "alignItems": "flex-start",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "Outfit-Medium",
              "fontSize": 14,
              "letterSpacing": 0.16,
              "lineHeight": 19,
              "marginLeft": 8,
              "marginRight": 16,
              "marginTop": 4,
              "paddingHorizontal": 4,
              "textAlign": "left",
            },
          ]
        }
      >
        label
      </Text>
      <Text
        style={
          [
            {
              "color": "#00A360",
              "fontFamily": "Outfit-Regular",
              "fontSize": 13,
              "letterSpacing": 0.16,
              "lineHeight": 18.2,
              "marginLeft": 8,
              "marginRight": 16,
              "marginVertical": 4,
              "paddingHorizontal": 4,
              "textAlign": "left",
            },
          ]
        }
      >
        Activadas
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignContent": "center",
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginTop": 0,
        },
      ]
    }
  >
    <View
      accessibilityRole="switch"
      accessibilityState={
        {
          "busy": undefined,
          "checked": 1,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      animatedStyle={
        {
          "value": {
            "backgroundColor": "#000000",
          },
        }
      }
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#000000",
          "borderRadius": 12,
          "padding": 4,
          "width": 44,
        }
      }
      testID="iconSwitchListItemSwitchId"
    >
      <View
        animatedStyle={
          {
            "value": {
              "transform": [
                {
                  "translateX": 20,
                },
              ],
            },
          }
        }
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "height": 16,
            "transform": [
              {
                "translateX": 20,
              },
            ],
            "width": 16,
          }
        }
      />
    </View>
  </View>
</View>
`;
